activation/activation.go

```diff
@@ -735,20 +756,61 @@ func (b *Builder) createAtx(
 		}
 	}
 
-	atx := wire.ActivationTxV1{
-		InnerActivationTxV1: wire.InnerActivationTxV1{
-			NIPostChallengeV1: *challenge,
-			Coinbase:          b.Coinbase(),
-			NumUnits:          nipostState.NumUnits,
-			NIPost:            wire.NiPostToWireV1(nipostState.NIPost),
-		},
-	}
-	if nonce != nil {
-		atx.VRFNonce = (*uint64)(nonce)
+	coinbase := b.Coinbase()
+
+	switch version {
+	case types.AtxV1:
+		atx := wire.ActivationTxV1{
+			InnerActivationTxV1: wire.InnerActivationTxV1{
+				NIPostChallengeV1: *wire.NIPostChallengeToWireV1(challenge),
+				Coinbase:          coinbase,
+				NumUnits:          nipostState.NumUnits,
+				NIPost:            wire.NiPostToWireV1(nipostState.NIPost),
+			},
+		}
+		if nonce != nil {
+			atx.VRFNonce = (*uint64)(nonce)
+		}
+		atx.Sign(sig)
+
+		return &atx, nil
+	case types.AtxV2:
+		atx := &wire.ActivationTxV2{
+			PublishEpoch:   challenge.PublishEpoch,
+			PositioningATX: challenge.PositioningATX,
+			Coinbase:       &coinbase,
```

Comment:
I thought we only include the coinbase if it is different from the previous ATX or if its the first ATX of that identity?

Reply:
Yes, that was the plan. I updated the code to fill coinbase only when it changed.